### PR TITLE
Prevent users from rating packages they authored

### PR DESCRIPTION
**Pull request by andrewmile**
This PR removes the ability for users to rate packages they have either authored or contributed to.  This change prevents the backend from accepting a rating by a package's author or contributor and the frontend will not display the components for rating a package if the authenticated user is the author or a contributor of the package.  This PR address concerns brought up in #10 

**Viewing a package authored by a different user**
image

**Viewing a package authored or contributed by the current user**
image

In addition, this PR provides a `purge:self-authored-package-ratings` that will delete any ratings where either of the following are true: 
- The `user_id` of the rating is the same as the `author_id` of the rated package 
- The package being rated has a collaborator with the same `user_id` as the rating's `user_id`.  

Running the command will display the following output:
image

### CODE REVIEW

File: app/Http/Controllers/InternalApi/RatingsController.php [hunk 1]
```diff
@@ -13,7 +14,14 @@ class RatingsController extends Controller
 {
     public function store()
     {
-        auth()->user()->ratePackage(request('package_id'), request('rating'));
+        try {
+            auth()->user()->ratePackage(request('package_id'), request('rating'));
+        } catch (SelfAuthoredRatingException $e) {
+            return response([
+                'status' => 'error',
+                'message' => 'A package cannot be rated by it\'s author',
```
Comment (mattstauffer) [inline app/Http/Controllers/InternalApi/RatingsController.php]: no apostrophe on its :)

Comment (andrewmile) [reply app/Http/Controllers/InternalApi/RatingsController.php]: oops, fixed!

**Comment by andrewmile** [comment]
That's a great question.  I didn't have my head wrapped around the collaborator model when I did this but yeah maybe users should not be able to rate a package they collaborated on either.  I can work on adding that to this PR.

**Comment by andrewmile** [comment]
@mattstauffer I updated the PR to also not allow users to rate a package they contributed to.

File: app/Jobs/UserRatePackage.php [hunk 2]
```diff
@@ -50,4 +57,14 @@ public function handle()
...
     }
+
+    private function isSelfAuthored($package)
+    {
+        return (int) $package->author_id === (int) $this->userId;
+    }
+
+    private function isSelfContributed($package)
+    {
+        return $package->contributors->pluck('user_id')->contains($this->userId);
```
Comment (mattstauffer) [inline app/Jobs/UserRatePackage.php]: Could we up the performance a bit by doing `$package->contributors()->pluck('user_id')`? I vaguely think that `pluck` on an Eloquent query is equivalent to a (SELECT that thing then pluck that thing on the result) but I can't say 100% for sure.

Comment (andrewmile) [reply app/Jobs/UserRatePackage.php]: Looks good to me!  I just changed it.

**Comment by andrewmile** [comment]
Thanks @mattstauffer !  I just made the change you requested

**Comment by mattstauffer** [comment]
Thanks @andrewmile!

**Comment by mattstauffer** [comment]
image

**Comment by andrewmile** [comment]
I just opened PR #26 to clarify the difference between the two lines in the output of the `purge:self-authored-package-ratings` command.  When you ran it above the result was 0 `self-authored` package ratings and 86 `self-contributed` package ratings.

**Pull request merged**